### ISSUE TITLE
`__match_args__` is not added to dataclasses with no fields

### ISSUE THREAD

**Issue by hamdanal** [python/mypy#15748]
**Bug Report**

stubtest complains about missing `__match_args__ ` if the dataclass has no fields

**To Reproduce**

Consider this python file:
```python
from dataclasses import dataclass
from typing import ClassVar

@dataclass
class Good:
    x: ClassVar[bool]
    y: int = 0

@dataclass
class Bad:
    x: ClassVar[bool]
```

and the corresponding stub file
```python
from dataclasses import dataclass
from typing import ClassVar

@dataclass
class Good:
    x: ClassVar[bool]
    y: int = ...

@dataclass
class Bad:
    x: ClassVar[bool]
```

**Expected Behavior**

```
Success: no issues found in 1 module
```

**Actual Behavior**

```
error: t.Bad.__match_args__ is not present in stub
Stub: in file /tmp/st-bug/t.pyi
MISSING
Runtime:
()

Found 1 error (checked 1 module)
```

**Your Environment**

- Mypy version used: mypy 1.4.1 (compiled: yes)
- Mypy command-line flags: N/A
- Mypy configuration options from `mypy.ini` (and other config files): N/A
- Python version used: 3.10.6

**Comment by AlexWaygood** [comment]
I think this manifests as a stubtest false positive, but the underlying issue is probably that mypy isn't synthesising `__match_args__` attributes for dataclasses that have 0 fields

**Comment by hamdanal** [comment]
Could be from this condition https://github.com/python/mypy/blob/14e7768c3bd8d1164e887ce3becba3459ebcfaa4/mypy/plugins/dataclasses.py#L368

I am not sure why there is a check for attributes here

**Comment by AlexWaygood** [comment]
> Could be from this condition

Looks like it to me!

> I am not sure why there is a check for attributes here

Nor me... want to make a PR? :)

**Comment by hamdanal** [comment]
Alright, I'll give it a try.

**Comment by hamdanal** [comment]
Filed #15749

Seems like the condition was added in the PR #10191 that introduced support for the match statement. This is a huge PR so I didn't follow to see if there is a reason for it or it is just an oversight :)

**Comment by AlexWaygood** [comment]
At a guess, I'd imagine it was probably just an oversight :)
